File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/items/equipment/HeadEquipment.java
package de.limbusdev.guardianmonsters.guardians.items.equipment;

import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian;

/**
 * HeadEquipment
 *
 * @author Georg Eckert 2017
 */

public class HeadEquipment extends Equipment {

    public enum Type {
        HELMET, BRIDLE, MASK, HEADBAND,
    }

    private Type type;

    public HeadEquipment(String name, Type type, int PStr, int PDef, int MStr, int MDef, int Speed, int HP, int MP, int Exp) {
        super(name, BodyPart.HEAD, PStr, PDef, MStr, MDef, Speed, HP, MP, Exp);
        this.type = type;
    }

    public Type getType() {
        return type;
    }

    @Override
    public boolean equipable(AGuardian m) {
        if(super.equipable(m)) {
            return (m.getSpeciesData().getHeadType() == type);
        } else {
            return false;
        }
    }
}

File: GuardianMonsters/guardians/src/de/limbusdev/guardianmonsters/guardians/items/equipment/HeadEquipment.kt
package de.limbusdev.guardianmonsters.guardians.items.equipment

import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian

/**
 * HeadEquipment
 *
 * @author Georg Eckert 2017
 */

class HeadEquipment
(
        name: String,
        val type: Type,
        PStr: Int,
        PDef: Int,
        MStr: Int,
        MDef: Int,
        Speed: Int,
        HP: Int,
        MP: Int,
        Exp: Int
)
    : Equipment(name, BodyPart.HEAD, PStr, PDef, MStr, MDef, Speed, HP, MP, Exp)
{
    enum class Type
    {
        HELMET, BRIDLE, MASK, HEADBAND
    }

    override fun canBeEquipped(guardian: AGuardian): Boolean
    {
        return super.canBeEquipped(guardian) && guardian.speciesDescription.headType == type
    }
}